pkg/instance/service.go

```diff
@@ -389,3 +393,134 @@ func (s service) destroy(instance *model.Instance) error {
 	}
 	return nil
 }
+
+func (s service) DeployDeployment(token string, deployment *model.Deployment) error {
+	stacks, err := s.collectStacks(deployment)
+	if err != nil {
+		return err
+	}
+
+	deploymentGraph, err := stack.ValidateNoCycles(stacks)
+	if err != nil {
+		return err
+	}
+
+	stackNames, err := deploymentOrder(deploymentGraph)
+	if err != nil {
+		return err
+	}
+
+	for _, stackName := range stackNames {
+		stack, err := s.stackService.Find(stackName)
+		if err != nil {
+			return err
+		}
+
+		instance := deployment.FindInstanceByStackName(stackName)
+		if instance == nil {
+			return errdef.NewNotFound("failed to find instance for stack %q", stackName)
+		}
+
+		instanceParameters := instance.Parameters
+
+		// Add parameters not supplied by user
+		for name, stackParameter := range stack.Parameters {
+			if _, ok := instanceParameters[name]; !ok {
+				instanceParameter := model.DeploymentInstanceParameter{
+					ParameterName: name,
+					StackName:     stack.Name,
+				}
+
+				if stackParameter.DefaultValue != nil {
```

Comment:
We should return an error if `!stack.Parameters[name].Consumed`, there is no default and the user did not provide a value, right?

We need to be 100% sure that we have all parameters for each instance. If we don't we should not call `deploy` but return an error.